### PR TITLE
docs(guide/migration): add TOC and fix heading levels

### PR DESCRIPTION
**Pull request by gkalpak**

### CODE REVIEW

File: docs/content/guide/migration.ngdoc [hunk 1]
```diff
@@ -3,21 +3,34 @@
...
+Although we try to avoid breaking changes, there are some cases where it is unavoidable:
 
 * AngularJS has undergone thorough security reviews to make applications safer by default,
 which drives many of these changes.
 * Several new features, especially animations, would not be possible without a few changes.
 * Finally, some outstanding bugs were best fixed by changing an existing API.
 
+<hr />
+
+## Table of Contents
```
Comment (gkalpak) [inline docs/content/guide/migration.ngdoc]: Any alternative suggestions for `Table of Contents` ?

Comment (Narretz) [inline docs/content/guide/migration.ngdoc]: Just "Contents"?

Comment (gkalpak) [inline docs/content/guide/migration.ngdoc]: I'll take it :) Thx !

**Comment by Narretz** [comment]
Otherwise, LGTM

**Pull request closed without merging**